spog/api/src/server.rs

```diff
@@ -133,6 +134,38 @@ impl Server {
     }
 }
 
+#[derive(Debug, Display, Error, From)]
+pub enum Error {
+    #[display(fmt = "response error: {:?}", "_0")]
+    Response(StatusCode, String),
+    #[display(fmt = "request error: {:?}", "_0")]
+    Request(reqwest::Error),
+    #[display(fmt = "url parse error: {}", "_0")]
+    UrlParse(url::ParseError),
+    #[display(fmt = "authentication error: {}", "_0")]
+    AuthClient(AuthClientError),
+}
+
+impl actix_web::error::ResponseError for Error {
+    fn error_response(&self) -> HttpResponse {
+        let mut res = HttpResponse::build(self.status_code());
+        res.insert_header(ContentType::plaintext());
+        match self {
+            Self::Response(_, error) => res.body(error.to_string()),
```

Comment:
I've been using the `ErrorInformation` struct for return data. There's also some logic in the frontend to extract some "useful" information. I think it make using that here too.

Reply:
Good idea, thanks.